doc/Plugin-API.md

```diff
@@ -167,14 +176,45 @@ The `Ext` side then gets the cached object, executes appropriate functions and r
+The lifecycle of these kinds of plugins is bound to _frontend connections_: for each connected frontend, the Theia backend spawns a plugin host host process dedicated to it in which these plugins are loaded and activated (as applicable to their declared activation events).
+In the plugin host for a frontend connection these plugins have access to Theia services as discussed above, isolated from Theia itself and from all of the other instances of the same plugin running in plugin hosts for all other frontend connections.
+
+Headless plugins, by contrast, are quite different in most respects:
+
+- they are encapsulated in a single plugin host process in the backend, separate from all frontend-connection plugin hosts.
+This host is spun up only if there are any headless plugins to run in it
+- Theia does not export any default API object, analogous to `vscode` or `theia` for other plugins, as Theia itself defines no use cases for headless plugins.
+Such use cases are entirely defined by the Theia-based application's requirements
+- Theia does not support any contribution points for headless plugins not any non-trivial activation events (only `'*'` and `'onStartupFinished'`).
+This is a corollary of the use cases being entirely application-specific: the application needs to define its own contribution points and activation events
+
+Thus, headless plugins are best suited to the contribution of third-party extensions of a Theia application's custom backend services, where those service are shared by all connected frontends or serve some kind of headless scenario like a CLI.
+
+A headless plugin may be restricted to only the headless deployment, in which case it may make this explicit by declaring the `"theiaHeadlessPlugin"` engine in its package manifest.
+Alternatively, a VS Code or Theia plugin that extends the frontend user experience may also contribute a headless entrypoint for a headless deployment by identifying such entrypoint script in the `"headless"` property of the `"theiaPlugin"` object in its package manifest in addition to the `"main"` entrypoint (for VS Code plugins) or the `"theiaPlugin.backend"` entrypoint (for Theia plugins).
+
+The only API namespaces that are available to headless plugins are those custom APIs that are contributed by the application's custom build-time extensions or by the activation of other headless plugins.
+For details of how to contribute custom API, see the [pertinent documentation](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/doc/how-to-add-new-custom-plugin-api.md).
+
+## Dependency Injection
+
+Both the `Main` and the `Ext` sides of the plugin API are configured using [InversifyJS](https://inversify.io) dependency injection.
+On the `Main` side, the usual Theia container modules are used to bind implementations of the API objects.
```

Comment:
Could you please add the specific symbol for that binding?

Reply:
Which specific symbol? There are loads of symbols that can be bound. What I'm trying to get at with this statement is that support for Inversify DI on the Main side is exactly like all of the other DI in Theia, using ContainerModules registered in package manifests and loaded at start-up by a code-generated script. Maybe I should just state it in those terms in the doc 😀